src/phl/ecPhlEqobs.ml

```diff
@@ -198,6 +198,23 @@ let rec s_eqobs_in_rev rsl rsr sim local (eqo:Mpv2.t) =
 
 and i_eqobs_in il ir sim local (eqo:Mpv2.t) =
   match il.i_node, ir.i_node with
+  | Sasgn(LvTuple lvls as lvl, ({e_node = Etuple els} as el)), Sasgn(LvTuple lvrs as lvr, ({e_node = Etuple ers} as er)) ->
+    let unpack lvs es =
+        let mk_assigns = List.map2 (fun lv e -> mk_instr (Sasgn (LvVar lv, e))) in
+        let lvs' = List.mapi (
+            (* These new variables will disappear after s_eqobs_in thanks
+               to the two consecutive assignments. *)
+            fun i (_, ty) -> PVloc (Format.sprintf " i_eqobs_in_tup_tmp%i" i), ty
```

Comment:
Using a non-syntaticaly valid variable name would ensure that you are collision free (e.g. by starting the name with a `$`)

Reply:
We start with a space, that should be equivalent, right ? But maybe `$`  is more explicit ?

Reply:
Yes, please. Something more visual.

Reply:
Changed ` ` to `$`.